src/Core/Managed/Shared/Extensibility/Metrics/MetricExtractorTelemetryProcessorBase.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Globalization;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+namespace Microsoft.ApplicationInsights.Extensibility.Metrics
+{
+    public abstract class MetricExtractorTelemetryProcessorBase : ITelemetryProcessor, ITelemetryModule, IDisposable
+    {
+        private MetricManager _metricManager = null;
```

Comment:
@macrogreg we have a versioned set of expected preaggs that should be documented. It is not a free-form list of aggs.